banner-plugin/src/admin/routes/banners/page.tsx

```diff
@@ -1,15 +1,173 @@
+import { useForm, FormProvider, Controller } from "react-hook-form";
+import { Button, FocusModal, Heading, Label, Input, Switch } from "@medusajs/ui";
 import { defineRouteConfig } from "@medusajs/admin-sdk";
-import {StackPerspective} from "@medusajs/icons";
+import { StackPerspective } from "@medusajs/icons";
+import { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+
+const bannerSchema = z.object({
+  title: z.string().min(1, "Title is required"),
+  link: z.string().min(1, "Link is required"),
+  image: z.string().min(1, "Image is required"),
+  image_mobile: z.string().nullable(),
+  sort_order: z.number().default(0),
+  enabled: z.boolean().default(true),
+});
+
+type BannerFormData = z.infer<typeof bannerSchema>;
 
 const BannersPage = () => {
-    return (
-        <div></div>
-    );
-}
+  const form = useForm<BannerFormData>({
+    defaultValues: {
+      title: "",
+      link: "",
+      image: "",
+      image_mobile: null,
+      sort_order: 0,
+      enabled: true,
+    },
+    resolver: zodResolver(bannerSchema),
+  });
+
+  const handleSubmit = async (data: BannerFormData) => {
+    try {
+      // TODO: Implement banner creation logic
+      console.log("Form data:", data);
+    } catch (error) {
+      console.error("Error creating banner:", error);
+    }
+  };
+
+  return (
+    <FocusModal>
+      <FocusModal.Trigger asChild>
+        <Button>Create Banner</Button>
+      </FocusModal.Trigger>
+      <FocusModal.Content>
+        <FormProvider {...form}>
+          <form onSubmit={form.handleSubmit(handleSubmit)} className="flex h-full flex-col overflow-hidden">
+            <FocusModal.Header>
+              <div className="flex items-center justify-end gap-x-2">
+                <FocusModal.Close asChild>
+                  <Button size="small" variant="secondary">
+                    Cancel
```

Comment:
```suggestion
                    {t("actions.cancel")}
```

that will require the following:

`const {t} = useTranslation()`